app/services/charges/charge_models/percentage_service.rb

```diff
@@ -13,6 +13,34 @@ def compute_amount
         compute_percentage_amount + compute_fixed_amount
       end
 
+      def amount_details
+        paid_units = units - free_units_value
+        paid_units = 0 if paid_units.negative?
+        percentage_rate_unit_amount = paid_units.zero? ? 0 : compute_percentage_amount.fdiv(paid_units)
+        free_events = aggregation_result.count >= free_units_count ? free_units_count : [aggregation_result.count - free_units_count, 0].max
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.